app/client/src/layoutSystems/anvil/editor/AnvilEditorDetachedWidgetOnion.tsx

```diff
@@ -23,7 +24,11 @@ export const AnvilEditorDetachedWidgetOnion = (props: BaseWidgetProps) => {
   useObserveDetachedWidget(props.widgetId);
   useHandleDetachedWidgetSelect(props.widgetId);
   useAddBordersToDetachedWidgets(props.widgetId, props.type);
-
+  useAnvilDetachedWidgetsDnD(
+    props.widgetId,
+    props.layout[0].layoutId,
+    !!props.isVisible,
```

Comment:
Shouldn't it be just `props.isVisible` without `!!`? Why do we need a double inversion here if it should be the  Boolean type anyway?

```suggestion
    props.isVisible,
```

Reply:
the type for isVisible is boolean|undefined since its not available for all widgets, so we had to use the `!!`

Reply:
Why is it unavailable for some widgets? Can we pass isVisible: false explicitly?

Reply:
The issue here is that we are using the same type definitions for fixed and anvil. so by definition this prop can be undefined, why coz there are canavs widgets in fixed that dont have a isVisible property not sure if there are any more widgets.

as far as setting false explicitly would be a clean up task across the fixed layout paradigm.
instead of doing this, the best thing would be to split withWidgetProps and also have separated type definitions for Anvil. we do have a tech debt task listed for this.

Reply:
Understood. There should be a link to BaseWidget task 🙂

Reply:
#28401